src/command_buffer.rs

```diff
@@ -0,0 +1,217 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0, <LICENSE-APACHE or
+// http://apache.org/licenses/LICENSE-2.0> or the MIT license <LICENSE-MIT or
+// http://opensource.org/licenses/MIT>, at your option. This file may not be
+// copied, modified, or distributed except according to those terms.
+
+use core::any::TypeId;
+use core::ptr::{self, NonNull};
+
+use crate::alloc::alloc::{alloc, dealloc, Layout};
+use crate::alloc::vec::Vec;
+use crate::archetype::TypeInfo;
+use crate::Entity;
+use crate::World;
+use crate::{align, DynamicBundle};
+
+/// Allows spawn operations to be buffered for future application to a ['World']
+///
+/// ```
+/// # use hecs::*;
+/// let mut world = World::new();
+/// let reserved_ent = world.reserve_entity();
+/// let mut entity_buffer = CommandBuffer::new();
+/// entity_buffer.spawn_at(reserved_ent,(true,0.10));
+/// entity_buffer.run_on(&mut world); // buffer can now be reused
+/// assert!(world.contains(reserved_ent));
+/// ```
+pub struct CommandBuffer {
+    entity: Vec<EntityIndex>,
```

Comment:
`entity` is singular. Because this stores information about multiple entities, it should be a plural, e.g. `entities`.